src/Raven.Server/Documents/TimeSeries/TimeSeriesPolicyRunner.cs

```diff
@@ -0,0 +1,683 @@
+                    database.Name,
+                    $"retention policy runner error in {database.Name}", msg,
+                    AlertType.RevisionsConfigurationNotValid, NotificationSeverity.Error, database.Name));
+
+                var logger = LoggingSource.Instance.GetLogger<TimeSeriesPolicyRunner>(database.Name);
+                if (logger.IsOperationsEnabled)
+                    logger.Operations(msg, e);
+
+                return null;
+            }
+        }
+        public static readonly Slice RawPolicySlice;
+
+        private static readonly TableSchema RollUpSchema;
+        public static readonly Slice TimeSeriesRollUps;
+        private static readonly Slice RollUpKey;
+        private static readonly Slice NextRollUpIndex;
+        private enum RollUpColumn
+        {
+            // documentId/Name
+            Key,
+            Collection,
+            NextRollUp,
+            PolicyToApply,
+            Etag,
+            ChangeVector
+        }
+
+        internal class RollUpState
+        {
+            public Slice Key;
+            public string DocId;
+            public string Name;
+            public long Etag;
+            public string Collection;
+            public DateTime NextRollUp;
+            public string RollUpPolicy;
+            public string ChangeVector;
+        }
+
+        static TimeSeriesPolicyRunner()
+        {
+            using (StorageEnvironment.GetStaticContext(out var ctx))
+            {
+                Slice.From(ctx, "TimeSeriesRollUps", ByteStringType.Immutable, out TimeSeriesRollUps);
+                Slice.From(ctx, "RollUpKey", ByteStringType.Immutable, out RollUpKey);
+                Slice.From(ctx, "NextRollUpIndex", ByteStringType.Immutable, out NextRollUpIndex);
+                Slice.From(ctx, RawTimeSeriesPolicy.PolicyString, ByteStringType.Immutable, out RawPolicySlice);
+            }
+
+            RollUpSchema = new TableSchema();
+            RollUpSchema.DefineKey(new TableSchema.SchemaIndexDef
+            {
+                StartIndex = (int)RollUpColumn.Key,
+                Count = 1, 
+                Name = RollUpKey
+            });
+
+            RollUpSchema.DefineIndex(new TableSchema.SchemaIndexDef // this isn't fixed-size since we expect to have duplicates
+            {
+                StartIndex = (int)RollUpColumn.NextRollUp, 
+                Count = 1,
+                Name = NextRollUpIndex
+            });
+        }
+
+        public unsafe void MarkForPolicy(DocumentsOperationContext context, Slice key, string collection, string name, long etag, DateTime baseline, string changeVector)
+        {
+            var config = Configuration.Collections[collection];
+
+            if (config.Disabled)
+                return;
+
+            var current = config.GetPolicyByTimeSeries(name);
+            if (current == null) // policy not found
+                // TODO: delete this time-series if not the raw one?
```

Comment:
So changing the configuration should keep left-overs?
The user should manually delete them?

Reply:
Removing the policy will keep the data by default and it will be user's responsibility to delete them.
 
Changing the parameters for an existing policy will take effect from the time of the change and will keep previous data intact. (unless is should re-aggregate due to the parameters change)